app/policies/action_policy.rb

```diff
@@ -1,13 +1,13 @@
 class ActionPolicy < ApplicationPolicy
-
-  ADMIN_OPERATIONS     = [Action::MEMO_OPERATION, Action::APPROVE_OPERATION, Action::DENY_OPERATION, Action::CANCEL_OPERATION].freeze
-  APPROVER_OPERATIONS  = [Action::MEMO_OPERATION, Action::APPROVE_OPERATION, Action::DENY_OPERATION].freeze
-  REQUESTER_OPERATIONS = [Action::CANCEL_OPERATION].freeze
-
   class Scope < ApplicationPolicy::Scope
     def resolve
       raise Exceptions::NotAuthorizedError, "Read access not authorized for #{scope}" unless permission_check('read', scope.model)
 
+      if user.params[:request_id]
```

Comment:
@hsong-rh If user.params is nil that if statement might fail. Most of the time for index call you will have params from the Controller. You might want to raise an Arguments error if its missing.